ENIGMAsystem/SHELL/Graphics_Systems/General/GSmodel.cpp

```diff
@@ -15,15 +17,651 @@
 *** with this code. If not, see <http://www.gnu.org/licenses/>
 **/
 
+#include "GSmodel_impl.h"
 #include "GSmodel.h"
+#include "GSvertex.h"
+#include "GSvertex_impl.h"
+#include "GSmatrix.h"
+#include "GStextures.h"
 #include "GSprimitives.h"
 
+#include "Universal_System/fileio.h"
+
 #include <math.h>
 
 using namespace std;
 
+namespace enigma {
+
+vector<Model*> models;
+extern unsigned char currentcolor[4];
+
+unsigned int split(const std::string &txt, std::vector<std::string> &strs, char ch)
+{
+	size_t pos = txt.find( ch );
+	size_t initialPos = 0;
+	strs.clear();
+
+	// Decompose statement
+	while( pos != std::string::npos ) {
+		strs.push_back( txt.substr( initialPos, pos - initialPos + 1 ) );
+		initialPos = pos + 1;
+
+		pos = txt.find( ch, initialPos );
+	}
+
+	// Add the last one
+	strs.push_back( txt.substr( initialPos, std::min( pos, txt.size() ) - initialPos + 1 ) );
+
+	return strs.size();
+}
+
+//split a string and convert to float
+vector<float> float_split(const string& str, const char& ch) {
+  string next;
+  vector<float> result;
+
+  for (string::const_iterator it = str.begin(); it != str.end(); it++)
+  {
+    if (*it == ch) {
+      if (!next.empty()) {
+        result.push_back(atof(next.c_str()));
+        next.clear();
+      }
+    } else {
+      next += *it;
+    }
+  }
+  if (!next.empty())
+    result.push_back(atof(next.c_str()));
+  return result;
+}
+
+//obj model parsing functions
+void string_parse(string *s) {
+  size_t spaces = 0;
+  bool trimmed = false;
+  bool checknormal = false;
+  for (unsigned int i = 0; i < s->size() ; i++)
```

Comment:
I can fix it, but this was not my code, it was originally written by Tizzio on the forums. Obviously it has never been touched but I believe it has remained working because various people have claimed to load obj models with ENIGMA over the years. I did not really want to touch it and probably won't even if you beg me. I had to touch at least a little to make it call d3d_model_vertex instead of the deleted `AddVertex` on `Model` struct.